libflux/src/flux/lib.rs

```diff
@@ -603,10 +603,56 @@ mod tests {
     use core::parser::Parser;
     use core::semantic::convert::convert_file;
     use core::semantic::env::Environment;
+    use core::semantic::fresh::Fresher;
     use core::semantic::nodes::infer_file;
-    use core::semantic::types::MonoType;
+    use core::semantic::types::{MonoType, Row, Tvar, TvarMap};
     use core::{ast, semantic};
 
+    pub struct MonoTypeNormalizer {
+        tv_map: TvarMap,
+        f: Fresher,
+    }
+
+    impl MonoTypeNormalizer {
+        pub fn new() -> Self {
+            Self {
+                tv_map: TvarMap::new(),
+                f: Fresher::default(),
+            }
+        }
+
+        pub fn normalize(&mut self, t: &mut MonoType) {
+            match t {
+                MonoType::Var(tv) => {
+                    let v = self.tv_map.entry(*tv).or_insert(self.f.fresh());
```

Comment:
This looks a bit odd. Won';t this mean we call `self.f.fresh` everytime we see a type variable even it is already been used? And that means that in those cases the type variable will be skipped.

Maybe a test to uses the same type variable more than once is needed to validate this behavior. 

I think we want the normalizer to be consistent and not skip any type variable numbers in the sequence.

Reply:
Oh good catch, I meant to pass in a closure using `or_insert_with()`